detection_rules/exception.py

```diff
@@ -4,10 +4,13 @@
 # 2.0.
 """Rule exceptions data."""
 from dataclasses import dataclass
+from datetime import datetime
 from pathlib import Path
-from typing import List, Literal, Optional, Union
+from typing import List, Literal, Optional, Union, Tuple
```

Comment:
up to you if it's worth having it in this PR, but wanted to note that since we're [relying](https://github.com/elastic/detection-rules/blob/main/pyproject.toml#L6) on python 3.12, we can simplify these:
- simple `dict`/`list`/`tuple` types
- `X | Y` instead of Union

Reply:
Since this PR introduced Tuple, I switched it to tuple, but I think updating the rest should happen in other PRs. That being said, 100% agree we should update.